peps/pep-9999.rst

```diff
@@ -0,0 +1,233 @@
+
+Rejected Alternatives
+--------------------------------------------
+Many alternative syntaxes have been proposed however no syntax other than ``f(=x)`` or ``f(x=)`` has garnered significant support. We here enumerate some of the most popular proposed alternatives and why we ultimately reject them.
+
+``f(=x)``
+'''''''''''''''''''''''''''''''''''''''''''''
+In favour of this form:
+
+* It is more similar to the established ``*args`` and ``**kwargs`` syntax for function calls
+* It draws more attention to itself when arguments are arranged vertically. In particular, if the arguments are of different lengths it is harder to find the equal sign at the end. Moreover, since Python is read left to right, the use of this feature is clearer to the reader earlier on.
+
+On the contrary:
+
+* While the prefix version is visually louder, in practice, there is no need for this feature to shout its presence any more than a typical named argument. By the time we read to the ``=`` it is clear that the value is filled in automatically just as the value is clear in the typical keyword argument case.
+* Semantically, this form communicates 'here is a value, fill in the parameter' which is not what we want to convey.
+* Less similar to f-string syntax
+* Less obvious that arbitrary expressions are invalid, e.g.`` f(=a+b)``
+
+
+``f(%x)`` or ``f(:x)`` or ``f(.x)``
+'''''''''''''''''''''''''''''''''''''''''''''
+Several flavours of this syntax have been proposed with the prefix form substituting another character for ``=``. However, no such form has gained traction and choice of symbol seems arbitrary compared to ``=``. Additionally, there is less precedent in terms of existing language features (such as f-string) or other languages (such as Ruby).
+
+
+``f(a, b, *, x)``
+'''''''''''''''''''''''''''''''''''''''''''''
+On a few occasions the idea has been floated to borrow the syntax from keyword-only function definitions. This is less arbitrary than ``f(%x)`` or variants, but no less so than ``f(x=)``. 
+
+However, we object that:
+
+* For any given argument, it is less clear from local context whether it is positional or named. The ``*`` could easily be missed in a long argument list and named arguments may be read as positional or vice versa.
+* It is unclear whether keyword arguments for which the value was not elided may follow the ``*``. If so, then their relative position will be inconsistent but if not, then an arbitrary grouping is enforced between different types of keyword arguments.
+
+
+Objections
+==================
+
+There are only a few hard objections to the introduction of this syntactic sugar. Most of those not in favour of this feature are simply in the camp of 'I wouldn't use it'. However over the extensive conversations about this feature, the following objections were the most common:
+
+The syntax is ugly
+--------------------------------------------
+This objection is by far the most common. On the contrary, we argue that:
```

Comment:
may want a less argumentative term than 'argue', e.g. 'believe' or 'contend'.

Reply:
While there's a lovely pun in calling the word `argue` "argumentative", it's also not too much of a problem here (eg imagine a mathematician saying "in the debate about pi vs tau, we argue that the elegance of measuring a circle by its diameter outweighs the elegance of radial measures being proportions of an entire turn").

Reply:
Fair enough, thanks